annotator/dirstral_annotator/recognizers/faces.py

```
@@ -35,7 +37,68 @@
+
+
+def _session_providers(app: object) -> list[str]:
+    """Providers the underlying ONNX sessions actually bound, best effort."""
+    try:
+        models = getattr(app, "models", {}) or {}
+        for model in models.values():
+            session = getattr(model, "session", None)
+            if session is not None:
+                return list(session.get_providers())
+    except Exception:  # pragma: no cover - diagnostics must never break a run
+        pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)